### PR TITLE
Add template for the unified changelog

### PR DESCRIPTION
**Pull request by monicasarbu**

### CODE REVIEW

**Comment by monicasarbu** [comment]
Please don't merge it yet, as I would like to reuse the same PR for adding a section for each past release.

**Comment by ruflin** [comment]
LGTM

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+=== Filebeat
+
+=== Winlogbeat
+
+// End Template
+
+
+// History of the past Beats releases
+
+= Beats version 1.0.0
```
Comment (ruflin) [inline CHANGELOG.asciidoc]: We should still have the link to the diff here as we had in the CHANGELOG.md

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+=== Topbeat
+- Rename timestamp field with @timestamp for a better integration with
+Logstash. #80
+
+=== Filebeat
+- Rename the timestamp field with @timestamp #168
+- Rename tail_on_rotate prospector config to tail_files
+- Removal of line field in event. Line number was not correct and does not add value. #217
+
+=== Winlogbeat
```
Comment (ruflin) [inline CHANGELOG.asciidoc]: I would remove the empty titles in the old one. So if there were no bugfixes etc, we remove it (same as we did with CHANGELOG.md)

**Comment by tsg** [comment]
We have to think how we can include this in the docs. My favorite solution would be to just do `include ../../CHANGELOG.asciidoc` in `libbeat/docs/index.asciddoc`, so there's no conversion/generation step. But this means that we have to delete or somehow comment out the template at the top and remove the empty sections at release time, like @ruflin suggested.

**Comment by ruflin** [comment]
@tsg It seems like asciidocs supports to include only parts of a file with tags: http://asciidoctor.org/docs/user-manual/#selecting-parts-of-a-document-to-include Didn't test it ...

File: CHANGELOG.asciidoc (hunk 1 above)
Comment (monicasarbu) [inline CHANGELOG.asciidoc]: done

File: CHANGELOG.asciidoc [hunk 3]
```diff
@@ -0,0 +1,435 @@
+
+// Start Template
+
+= Beats version x.x.x
```
Comment (ruflin) [inline CHANGELOG.asciidoc]: Here you can use the diff from the last release to HEAD

File: CHANGELOG.asciidoc (hunk 1 above)
Comment (ruflin) [inline CHANGELOG.asciidoc]: The one for 1.0.0 and head seems to be missing.

File: CHANGELOG.asciidoc (hunk 2 above)
Comment (monicasarbu) [inline CHANGELOG.asciidoc]: done

**Comment by monicasarbu** [comment]
I created a Release notes page that includes at the beginning links to all past releases and then the release notes for each past release.

File: libbeat/docs/release.asciidoc [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+This section summarizes the changes in each release.
+
+* <<release-notes-1.0.0>>
+* <<release-notes-1.0.0-rc2>>
+* <<release-notes-1.0.0-rc1>>
+* <<release-notes-1.0.0-beta4>>
+
+--
+
+include::{beats-dir}/CHANGELOG.asciidoc[]
```
Comment (ruflin) [inline libbeat/docs/release.asciidoc]: So this only includes the most recent release? Wouldn't we have then to specific the "tag"? http://asciidoctor.org/docs/user-manual/#selecting-parts-of-a-document-to-include (I didn't test if it works).

**Comment by dedemorton** [comment]
@debadair Wondering at this point if we should be aiming for consistency in the structure/content of release notes across the Elastic product documentation?

**Comment by monicasarbu** [comment]
@dedemorton @ruflin  I was trying to find a simpler way to have something similar with ES, but It seems that it's more work to do for us at each release. They have a release notes directory with all release notes, a release notes file for each release. To achieve this, we will need to create a release notes file for each release. I am not sure if this is what we want ...
I agree that the current implementation is not a very nice solution, but it requires no extra changes for each release.

**Comment by monicasarbu** [comment]
I propose to keep the current implementation and open this subject for discussion to find a better solution that we can then later implement in a different PR.

**Comment by monicasarbu** [comment]
After thinking a bit more, let me play a bit more with asciidoc.

**Comment by dedemorton** [comment]
@monicasarbu Makes sense. We might even be able to make some minor tweaks to the text without disrupting the structure, but I'll leave that discussion for another time.

**Comment by dedemorton** [comment]
@monicasarbu You might want to wait for @debadair to respond because she might have a vision for where we're going with Release Notes that's not reflected in the current structure. I just wanted to ask the question to get us started thinking about it, especially since we are moving the content to the doc.

**Comment by monicasarbu** [comment]
@dedemorton yes, you are right. It might be that there is a settings tweet to make it look like ES.

**Comment by ruflin** [comment]
@monicasarbu Based on the docs of asciidocs I posed above, we could implement a solution with a latest tag, so we would never have to change it. Did you try it out if the including tags work?

**Comment by monicasarbu** [comment]
@ruflin No, I am trying to achieve this: https://www.elastic.co/guide/en/elasticsearch/reference/2.1/es-release-notes.html, generate a page for each past release notes.

**Comment by monicasarbu** [comment]
@ruflin The problem that I have is not that I cannot include only a section from the changelog. It's that I might need to create a separate file for each release notes. This is what ES is currently doing. This is something that I am trying to avoid and trying to see if I can find a easier solution.

**Comment by ruflin** [comment]
@monicasarbu Got it. Worst case is we still have to generate the file but because we know the content which is just an include wit the tag, we could perhaps generate it. In case I have an idea how we can even get rid of the file, I will let you know.

**Pull request merged**